reset.d.ts

```diff
@@ -0,0 +1,9 @@
+type NonFalsy<T> = T extends false | 0 | '' | null | undefined | 0n ? never : T;
+
+interface Array<T> {
+  filter<S extends T>(predicate: BooleanConstructor, thisArg?: any): NonFalsy<S>[];
+}
+
+interface ReadonlyArray<T> {
+  filter<S extends T>(predicate: BooleanConstructor, thisArg?: any): NonFalsy<S>[];
+}
```

Comment:
we have `ts-reset` in the BX, this is just the array filter util from that
like
```ts
[false, 1,2, undefined, 3].filter(Boolean) 
```
ts default behavior is to keep the type `number | false | undefined`
with this change it narrows it down to `number` as we know it can't be "falsy" values because of the .filter(Boolean)

Reply:
so does this basically prevent the need to do `[false, 1,2, undefined, 3].filter((item): item is number => item)`?

Reply:
so does this basically prevent the need to do [false, 1,2, undefined, 3].filter((item): item is number => item)?

Reply:
yep